criu/cr-check.c

```diff
@@ -1551,3 +1553,125 @@ static char *feature_name(int (*func)(void))
 	}
 	return NULL;
 }
+
+#ifndef CAP_CHECKPOINT_RESTORE
+#define CAP_CHECKPOINT_RESTORE 40
+#endif
+
+static bool has_capability(int cap, struct vfs_cap_data *cap_data)
+{
+	int mask = CAP_TO_MASK(cap);
+	int index = CAP_TO_INDEX(cap);
+	u32 inheritable;
+	u32 permitted;
+
+	inheritable = le32toh(cap_data->data[index].inheritable);
+	permitted = le32toh(cap_data->data[index].permitted);
+
+	if (!(mask & permitted)) {
+		pr_debug("Capability %d not permitted\n", cap);
+		return false;
+	}
+
+	if (!(mask & inheritable)) {
+		pr_debug("Capability %d not inheritable\n", cap);
+		return false;
+	}
+
+	return true;
+}
+
+static int pr_set_dumpable(int value)
+{
+	int ret = prctl(PR_SET_DUMPABLE, value, 0, 0, 0);
+	if (ret < 0)
+		pr_perror("Unable to set PR_SET_DUMPABLE");
+	return ret;
+}
+
+int check_uid(void)
+{
+	const char *attribute = "security.capability";
+	const char *criu = "/proc/self/exe";
```

Comment:
I have to lol at this. Didn't we just argue that it's a bad idea to rely on exe links. :D